built_json/lib/src/built_map_serializer.dart

```diff
@@ -11,46 +11,61 @@ class BuiltMapSerializer implements Serializer<BuiltMap> {
   final String wireName = 'map';
 
   @override
-  Object serialize(Serializers serializers, BuiltMap object,
-      {GenericType genericType: const GenericType()}) {
-    final keyTypes = genericType.parameters.isEmpty
-        ? const GenericType()
-        : genericType.parameters[0];
-    final valueTypes = genericType.parameters.isEmpty
-        ? const GenericType()
-        : genericType.parameters[1];
+  Object serialize(Serializers serializers, BuiltMap builtMap,
+      {FullType specifiedType: const FullType()}) {
+    final isUnderspecified =
+        specifiedType.isObject || specifiedType.parameters.isEmpty;
+
+    final keyTypes = specifiedType.parameters.isEmpty
```

Comment:
why are keyTypes and valueTypes plural? Can there be more than one key type or value type?